Compilation fails with a narrowing conversion error

When I try to compile on Fedora with C++11, it throws the following error:

```
bot_combat.cpp: At global scope:
bot_combat.cpp:88:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
   88 | };
      | ^
bot_combat.cpp:156:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  156 | };
      | ^
bot_combat.cpp:173:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  173 | };
      | ^
bot_combat.cpp:243:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  243 | };
      | ^
bot_combat.cpp:299:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  299 | };
      | ^
bot_combat.cpp:340:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  340 | };
      | ^
bot_combat.cpp:429:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  429 | };
      | ^
bot_combat.cpp:501:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  501 | };
      | ^
bot_combat.cpp:506:1: error: narrowing conversion of ‘0.0f’ from ‘float’ to ‘bool’ [-Wnarrowing]
  506 | };
```

It seems to be due to the fact that C++ doesn't narrow anymore. See https://www.stroustrup.com/C++11FAQ.html#narrowing
I'm not well versed in C++ so I could be wrong.

--- a/dlls/bot_combat.cpp
+++ b/dlls/bot_combat.cpp
@@ -84,7 +84,7 @@ const bot_weapon_select_t valve_weapon_select[] = {
     100, TRUE, 70, 1, 1,
 	FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_PISTOL},
     /* terminator */
-    {0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
+    {0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
 };
 
 const bot_weapon_select_t gearbox_weapon_select[] = {
@@ -152,7 +152,7 @@ const bot_weapon_select_t gearbox_weapon_select[] = {
 	 0.0f, 1200.0f, 0.0f, 1200.0f,
 	 100, TRUE, 70, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_PISTOL},
 	/* terminator */
-	{0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
+	{0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
 };
 
 const bot_weapon_select_t cs_weapon_select[] = {
@@ -169,7 +169,7 @@ const bot_weapon_select_t cs_weapon_select[] = {
 	 100, TRUE, 100, 1, 0,
 	 FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_PISTOL},
 	/* terminator */
-	{0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
+	{0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
 };
 
 const bot_weapon_select_t dod_weapon_select[] = {
@@ -239,7 +239,7 @@ const bot_weapon_select_t dod_weapon_select[] = {
 	 0.0f, 50.0f, 0.0f, 0.0f,
 	 100, TRUE, 100, 0, 0, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_MELEE},*/
 	/* terminator */
-	{0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
+	{0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
 };
 
 const bot_weapon_select_t tfc_weapon_select[] = {
@@ -295,7 +295,7 @@ const bot_weapon_select_t tfc_weapon_select[] = {
 	 40.0f, 600.0f, 0.0f, 0.0f,
 	 100, TRUE, 100, 1, 0, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f},
 	/* terminator */
-	{0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
+	{0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
 };
 
 const bot_weapon_select_t gunman_weapon_select[] = {
@@ -336,7 +336,7 @@ const bot_weapon_select_t gunman_weapon_select[] = {
 	0, true, 100, 0, 0,
 	false, false, false, false, 0.0f, 0.0f, WEAPON_NONE},
 	/* terminator */
-    {0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
+    {0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f, WEAPON_NONE}
 };
 
 // see AvHBasePlayerWeapon::mRange
@@ -425,7 +425,7 @@ const bot_weapon_select_t ns_weapon_select[] = {
 	FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f},
 
     /* terminator */
-    {0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
+    {0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
 };
 
 const bot_weapon_select_t hunger_weapon_select[] = {
@@ -497,12 +497,12 @@ const bot_weapon_select_t hunger_weapon_select[] = {
     100, TRUE, 70, 1, 1,
 	FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f},
     /* terminator */
-    {0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
+    {0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
 };
 
 const bot_weapon_select_t ship_weapon_select[] = {
 	/* terminator */
-	{0, "", 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
+	{0, "", 0, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0, TRUE, 0, 1, 1, FALSE, FALSE, FALSE, FALSE, 0.0f, 0.0f}
 };
 
 edict_t *BotFindEnemy( bot_t *pBot )

